### ISSUE TITLE
[Internal] Room - Room chat avatar is different for user who creates room & who has been invited to room

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#33470]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.15-4
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** Applause - Internal Team
**Slack conversation:**

## Action Performed:
1. Launch app
2. Tap fab--start chat
3. Tap room
4. Enter room name and select a workspace with custom avatar
5. Tap create room
6. Tap header
7. Tap members---Invite
8. Invite user X to room
9 .Navigate to LHN and note avatar of room chat
10. Go to https://staging.new.expensify.com/
11. Login as user X
12. Note avatar of room chat

## Expected Result:
Room chat avatar must be same for user who creates room and for user who has been invited to room

## Actual Result:
Room chat avatar is different for user who creates room and for user who has been invited to room

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Native
- [x] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

</details> 

https://github.com/Expensify/App/assets/78819774/fbc7a3bd-7ef8-4165-83c4-072d8c245c11

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~013499a52b4634b0ed</li>
        <li>Upwork Job ID: 1737991692071854080</li>
        <li>Last Price Increase: 2023-12-29</li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @grgia</details>

**Comment by unicorndev-727** [comment]
## Proposal
from: @unicorndev-727
### Please re-state the problem that we are trying to solve in this issue.
Room chat avatar is different for user who creates room & who has been invited to room
### What is the root cause of that problem?
The root cause is that the user who has been invited to room doesn't have the policy data about workspace in Onyx so that the default workspace avatar will be showed.
This issue only happens when we invite the user to the only room.
https://github.com/Expensify/App/blob/8970aa3cd4ffc73cc59b9de3a50fd3d4da361f15/src/libs/ReportUtils.ts#L1247-L1249

It is a screen shot of the room owner.
![image](https://github.com/Expensify/App/assets/75538817/8acfa58c-0873-4df4-becd-8ea5d52335a7)
It is a screen shot of the room invited user.
![image](https://github.com/Expensify/App/assets/75538817/a4edbfde-6e39-4f51-ab03-087be96bd7c0)
### What changes do you think we should make in order to solve the problem?
Backend should send the workspace data to the invited users when we invite users to the room.
### What alternative solutions did you explore? (Optional)
 N/A

**Comment by alexpensify** [comment]
@getusha - - when you get a chance, can you review if this proposal will fix this issue? Thanks! 

**Heads up, I will be offline from Friday, December 22, to Thursday, January 4, 2024. I will not be actively watching over this GitHub during that period. If anything urgent is needed here, please ask for help in the #expensify-open-source Slack Room-- thanks!**

**Comment by getusha** [comment]
@unicorndev-727 do you mind adding more details to the solution you mentioned in your proposal?

**Comment by unicorndev-727** [comment]
> they can still access the `policyAvatar` (beside `policyName`) to display as room avatar

In fact, this is the best place to call the backend API to allow workspace data to invited users who are not members of the workspace.
https://github.com/Expensify/App/blob/595bf4075988c0b64cb0db1e37694284637fbea2/src/libs/actions/Report.ts#L2165-L2175
Frontend is subscribted to policy data properly to get workspace icons here
https://github.com/Expensify/App/blob/595bf4075988c0b64cb0db1e37694284637fbea2/src/components/LHNOptionsList/LHNOptionsList.js#L117
and here 
https://github.com/Expensify/App/blob/595bf4075988c0b64cb0db1e37694284637fbea2/src/libs/ReportUtils.ts#L1270.
But it doesn't work because we don't have the workspace data in `ONYXKEYS.COLLECTION.POLICY`.
So we need to update `inviteToRoom` API to share workspace data with the invited users who aren't member of the workspace by setting role to `guest` in policy so that the guest can only read policy data like name, avatar and so on, I think.

```
type InviteToRoomParameters = {
        reportID: string;
        inviteeEmails: string[];
        policyId: string;
        role: string;
    };

    const parameters: InviteToRoomParameters = {
        reportID,
        inviteeEmails,
        policyId: report.policyId,
        role: 'guest'
    };

    API.write('InviteToRoom', parameters, {optimisticData, successData, failureData});
```
![image](https://github.com/Expensify/App/assets/75538817/c9f39950-bc6e-4f32-82b7-e3f7ab89a6b5)
![image](https://github.com/Expensify/App/assets/75538817/f20537f5-cc87-4c8d-9954-d8ae29b08588)
This prevents inconsistencies in policy updates and also ensures security.

**Comment by getusha** [comment]
Reviewing proposals.

**Comment by alexpensify** [comment]
@getusha - I'm back online again. Any update regarding the proposals?

**Comment by getusha** [comment]
🎀 👀 🎀 
Pulling up internal engineer to confirm if we can return workspace data (avatar) to a non-member user that joined a room and fix this on the backend initially.

**Comment by grgia** [comment]
@getusha looking into this

**Comment by grgia** [comment]
Internal convo link: https://expensify.slack.com/archives/C03TQ48KC/p1704412668387289

**Comment by alexpensify** [comment]
Thanks, @grgia for this update!

**Comment by alexpensify** [comment]
Looks like there is another PR to review here.

**Comment by alexpensify** [comment]
@getusha and @grgia - can we get your feedback here? Thanks!

**Comment by alexpensify** [comment]
@getusha or @grgia - any update here?

**Comment by alexpensify** [comment]
Thanks, so we will wait for feedback from @getusha. Thanks!

**Comment by lanitochka17** [comment]
While testing these steps, the team discovered that the internal avatars were different. The steps are the same.
Should we open a separate problem or relate it to this one?
Thanks

https://github.com/Expensify/App/assets/78819774/c9393cd6-ecf1-435f-a61e-9d26af2a6563

**Comment by alexpensify** [comment]
I agree to separate the GH issue, thanks!

**Comment by grgia** [comment]
@dukenv0307 if you'd like to open a new PR and do testing instead, that works as well. I can close my PR

**Comment by alexpensify** [comment]
Done!

**Comment by alexpensify** [comment]
Daily update: waiting for the PR

**Comment by getusha** [comment]
@dukenv0307 don't forget to check https://github.com/Expensify/App/pull/34115#issuecomment-1889007115

**Comment by getusha** [comment]
@grgia could you please check this https://github.com/Expensify/App/issues/33470#issuecomment-1926205427 on BE side? ty!

**Comment by alexpensify** [comment]
Weekly Update: On Hold, waiting for @grgia's feedback.

**Comment by alexpensify** [comment]
Weekly Update: I followed up in chat for help here.

**Comment by grgia** [comment]
@getusha could you please summarize where we're at with this issue and what questions are left outstanding. If there are BE changes required, I can make them

**Comment by getusha** [comment]
> @getusha could you please summarize where we're at with this issue and what questions are left outstanding. If there are BE changes required, I can make them

@grgia We have 2 issues related to this, more likely a BE bug.

1. https://github.com/Expensify/App/pull/34115#issuecomment-1889007115
2. https://github.com/Expensify/App/pull/35792#issuecomment-1930691489

github confused the PR linked to this issue as merged because another PR had commits it doesn't supposed to have, we've got a new PR now :)

**Comment by alexpensify** [comment]
Weekly Update: Waiting on feedback

**Comment by getusha** [comment]
@grgia any updates on these issues? https://github.com/Expensify/App/issues/33470#issuecomment-1957586801

**Comment by grgia** [comment]
I looked into this the other way it's been marked as internal but I don't think this is a high priority bug right now. What we will do is use the `policyName` pattern as @dukenv0307 suggested for sharing the avatar (if any) as well.

**Comment by grgia** [comment]
> the partial compensation sounds good to m
@getusha I agree with this

**Comment by grgia** [comment]
https://github.com/Expensify/Auth/pull/9708/files
For personal reference, this is where the policyAvatar was added. Looking to see if there's an issue with this flow.

@getusha @dukenv0307 what's being returned in your testing?

**Comment by grgia** [comment]
@getusha @dukenv0307 I got a chance to get dev set up and tested on your test branch and cannot replicate the bug, could you reiterate the problem? 

<img width="1293" alt="Screenshot 2024-03-06 at 4 48 46 PM" src="https://github.com/Expensify/App/assets/38015950/879890a9-b963-4595-9436-ac0dacf180c6">

**Comment by getusha** [comment]
@grgia 

1. Invite User B to a room
2. After user B is invited to the room
3. Change the avatar of the workspace linked to the room

The new avatar will not be synced that's the issue, it loads fine first time.

https://github.com/Expensify/App/assets/75031127/cf61eb46-1da1-4197-8bad-d07d77da8496

**Comment by alexpensify** [comment]
@grgia - I think you are back online this week, we need feedback here. Thanks!

**Comment by grgia** [comment]
@dukenv0307 @getusha could you rephrase your question? Could you catch me up on what is required? Is this internal only?

**Comment by grgia** [comment]
>  it seems like the policyAvatar is not updated when the user changes the workspace avatar

Which command? @dukenv0307

**Comment by alexpensify** [comment]
Weekly Update: Waiting for feedback here

**Comment by alexpensify** [comment]
Moving back to Weekly

**Comment by grgia** [comment]
@dukenv0307 what would we like to happen instead here?

**Comment by alexpensify** [comment]
## Heads up, I will be offline until Tuesday, May 28, 2024, and will not actively watch over this GitHub during that period.

If anything urgent is needed here, please ask for help in the #expensify-open-source Slack Room-- thanks!

**Comment by alexpensify** [comment]
Weekly Check-In: No update yet, we are waiting for feedback from @grgia

**Comment by alexpensify** [comment]
No update yet, this one is still a hot pick.

**Comment by getusha** [comment]
@alexpensify @grgia we have a PR raised by @dukenv0307 https://github.com/Expensify/App/pull/37003, should we proceed with that? the internal engineer can work on the BE issue which i think is a separate case.

**Comment by alexpensify** [comment]
I've asked to see if I can get another engineer's feedback here.

**Comment by techievivek** [comment]
There have been a lot of conversations already, so it would be great if someone could help me clarify the timeline:

1. Have we already fixed the backend changes to return the policy avatar when a non-workspace member is invited to the room?
2. Have we also updated the frontend to handle the above backend change?
3. Is our current issue that the room avatar doesn't change for invited users when we update the policy avatar?

Which step are we currently in, and what problem lies ahead? I am assigning myself to the issue so we can move this forward ASAP.

**Comment by techievivek** [comment]
> @alexpensify @grgia we have a PR raised by @dukenv0307 https://github.com/Expensify/App/pull/37003, should we proceed with that? the internal engineer can work on the BE issue which i think is a separate case.

Yeah, we can work on this parallel since there shouldn't be any breaking changes, I just wanted to understand what has already been done and what are pending things here.

**Comment by alexpensify** [comment]
I don't believe these backend changes have happened yet. @getusha and @dukenv0307 do you agree with this plan? Thanks!

**Comment by techievivek** [comment]
Gentle bump @getusha @dukenv0307 on my above question.

**Comment by techievivek** [comment]
@dukenv0307  Ok, great, I understand things correctly now. Can we move forward with the frontend changes then? I can work on the backend changes, but we don't need to HOLD the frontend PR for it since both are separate flow and one doesn't need to wait for the ther, thanks.

**Comment by alexpensify** [comment]
@dukenv0307 keep us posted if you can move forward with this plan - thanks!

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by alexpensify** [comment]
## Next Steps

@isabelastisser the discussion is ongoing here. I reassigned the `Bug` label to keep this one moving forward while I'm OOO. Thanks! 

## Heads up, I will be offline until Tuesday, September 3, 2024, and will not actively watch over this GitHub during that period. 

If anything urgent is needed here, please ask for help in the #expensify-open-source Slack Room-- thanks!

**Comment by isabelastisser** [comment]
Not overdue.

**Comment by isabelastisser** [comment]
Bump @dukenv0307 @getusha

**Comment by alexpensify** [comment]
Thanks, @isabelastisser, for your help! I'm online and taking this GH back.

**Comment by alexpensify** [comment]
@techievivek - we need your follow-up here. Thanks!

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (Second week)

**Comment by techievivek** [comment]
> I checked again and the bug is fixed on mweb on the latest main. Can the QA team check again? Correct me if I missed something.
QA team just confirmed that this is fixed so we can move forward with the payment here. @alexpensify

**Comment by alexpensify** [comment]
Payouts due: 2024-09-09

- [ ] Contributor: $500 @dukenv0307 
- [ ] Contributor+: $500 @getusha 

Upwork job is [here](https://www.upwork.com/jobs/~021833263964040656607).

**Comment by alexpensify** [comment]
@dukenv0307 - I sent an offer in Upwork. Please accept and I can complete the payment process. 

@getusha - please submit a request in Chat. 

Thanks!

**Comment by getusha** [comment]
@alexpensify i should get paid on Upwork as well since the issue was created before my ND payment eligibility date.

**Comment by alexpensify** [comment]
@getusha - thanks for flagging, I sent over an offer in Upwork.

**Comment by alexpensify** [comment]
@dukenv0307 - thanks, I completed the payment process in Upwork!

**Comment by getusha** [comment]
@alexpensify accepted the offer. ty!
